### ISSUE TITLE
Does NVIDIA H100-SXM 0x2330 support cc?

### ISSUE THREAD

**Issue by jianlinjiang** [NVIDIA/nvtrust#67]
I launch the confidential VM, however the QEMU complains with 
```
[  349.818805] NVRM: failed to initialize module.
[  349.902869] NVRM: The NVIDIA GPU 0000:01:00.0 (PCI ID: 10de:2330)
[  349.902869] NVRM: installed in this system is not supported by the
[  349.902869] NVRM: NVIDIA 550.90.07 driver release.
[  349.902869] NVRM: Please see 'Appendix A - Supported NVIDIA GPU Products'
[  349.902869] NVRM: in this release's README, available on the operating system
[  349.902869] NVRM: specific graphics driver download page at www.nvidia.com.
[  349.909275] nvidia: probe of 0000:01:00.0 failed with error -1
[  349.909739] NVRM: The NVIDIA probe routine failed for 1 device(s).
[  349.910219] NVRM: None of the NVIDIA devices were initialized.
``` 

Since I run the command `apt install nvidia-driver-550-server-open` to install the nvidia driver in the CVM, but the driver doesn't support our H100-SXM.

**Comment by Tan-YiFan** [comment]
Did you enable CC mode on the host?

**Comment by jianlinjiang** [comment]
> Did you enable CC mode on the host?


![Image](https://github.com/user-attachments/assets/7503254f-25f0-4444-a0e5-54d8736a0e28)


yes, I enabled. However, after I launched a VM, the gpu admin tool shows error, even I shutdown the VM.  If I reboot the host, the gpu admin tool returns normal output.



![Image](https://github.com/user-attachments/assets/244468ee-0f44-4cd3-b733-748d33ee9445)

**Comment by Tan-YiFan** [comment]
Did you execute `nvidia-persistenced --uvm-persistence-mode` in the guest before `nvidia-smi`?

**Comment by jianlinjiang** [comment]
> Did you execute `nvidia-persistenced --uvm-persistence-mode` in the guest before `nvidia-smi`?

yes, but I just modified the /usr/lib/systemd/system/nvidia-persistenced.service as the document.

**Comment by Tan-YiFan** [comment]
Did you check the VBIOS version?

**Comment by jianlinjiang** [comment]
> Did you check the VBIOS version?

no, nvidia-smi shows can't communicate with nvidia driver

**Comment by Tan-YiFan** [comment]
Please check the vbios version on the host

**Comment by moconnor725** [comment]
where is the GPU on the PCI bus?  I see references to 01:00.0 (qemu startup) and 04:00.0 (gpu admin tool).

what's the OS version, linux kernel, and OVMF versions?

**Comment by jianlinjiang** [comment]
> where is the GPU on the PCI bus? I see references to 01:00.0 (qemu startup) and 04:00.0 (gpu admin tool).
> 
> what's the OS version, linux kernel, and OVMF versions?

I followed the doc. host and guest's Os are ubuntu 22.04. linux kernel is 5.19. I have four H100 SXM GPUs. They are 03:00.0, 04:00.0, e3:00.0, e4:00.0. I choose 04:00.0 for the qemu parameters.

**Comment by jianlinjiang** [comment]
> Please check the vbios version on the host

 VBIOS Version                         : 96.00.51.00.0D

Is it too old? @Tan-YiFan

**Comment by Tan-YiFan** [comment]
According to https://docs.nvidia.com/cc-deployment-guide-snp.pdf, the minimum required VBIOS version is 96.00.5E.00.00.

For information about updating VBIOS, please contact the vendor. (Reference: https://forums.developer.nvidia.com/t/vbios-update-on-h100-intel-tdx-h100/302464/2)

**Comment by moconnor725** [comment]
The very first post had "01:00.0" which suggests you will need to modify the launch_vm.sh script to point to the correct GPU which look correct at  03:00.0, 04:00.0, e3:00.0, e4:00.0

As for the firmware, yes, the firmware is out of date.  Please contact your OEM.  They should have access to the latest set of firmware.

**Comment by amshinde** [comment]
Closing this issue due to inactivity. If you're still experiencing this or need further assistance, please feel free to reopen.
